src/Libraries/PythonNodeModelsWpf/PythonNode.cs

```diff
@@ -41,21 +42,21 @@ public void CustomizeView(PythonNode nodeModel, NodeView nodeView)
                 var pythonEngineVersionMenu = new MenuItem { Header = PythonNodeModels.Properties.Resources.PythonNodeContextMenuEngineSwitcher, IsCheckable = false };
                 nodeView.MainContextMenu.Items.Add(pythonEngineVersionMenu);
                 pythonEngine2Item.Click += UpdateToPython2Engine;
-                pythonEngine3Item.Click += UpdateToPython3Engine;
-                pythonEngineVersionMenu.Items.Add(pythonEngine2Item);
-                pythonEngineVersionMenu.Items.Add(pythonEngine3Item);
-
-                // Check the correct item based on NodeModel engine version
-                if (pythonNodeModel.Engine == PythonEngineVersion.IronPython2)
+                pythonEngine2Item.SetBinding(MenuItem.IsCheckedProperty, new Binding(nameof(pythonNodeModel.Engine))
                 {
-                    pythonEngine2Item.IsChecked = true;
-                    pythonEngine3Item.IsChecked = false;
-                }
-                else
+                    Source = pythonNodeModel,
+                    Converter = new EnumToBooleanConverter(),
```

Comment:
Can you put some comment here describing the default behavior in future for ppl to refer to? Thanks.

Reply:
I added a comment. Let me know if it captures your concerns @QilongTang